activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,24 @@
+*   Add support for finding records based on signed ids, which are tamper-proof, verified ids that can be
+    set to expire and scoped with a purpose. This is particularly useful for things like password reset 
+    or email verification, where you want the bearer of the signed id to be able to interact with the 
+    underlying record, but usually only within a certain time period.
+    
+    ```ruby
+    signed_id = User.first.signed_id expires_in: 15.minutes, purpose: :password_reset
+    
+    User.find_signed signed_id # => nil, since the purpose does not match
+    
+    travel 16.minutes
+    User.find_signed signed_id # => nil, since the signed id has expired
```

Comment:
```suggestion
    User.find_signed signed_id, purpose: :password_reset # => nil, since the signed id has expired
```

otherwise the reason of getting `nil` could have been that the purpose didn't match